src/dsp_tools/commands/xmlupload/models/ontology_diagnose_models.py

```diff
@@ -0,0 +1,102 @@
+# sourcery skip: use-fstring-for-concatenation
+
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import pandas as pd
+
+from dsp_tools.models.exceptions import UserError
+
+
+@dataclass(frozen=True)
+class Ontology:
+    """This class saves the properties and the classes from an ontology."""
+
+    classes: list[str] = field(default_factory=list)
+    properties: list[str] = field(default_factory=list)
+
+
+@dataclass
+class OntoCheckInformation:
+    """This class saves information that is needed to check if consistency with the ontology."""
+
+    default_ontology_prefix: str
+    onto_lookup: dict[str, Ontology]
+    save_location: Path
+
+
+@dataclass(frozen=True)
+class InvalidOntologyElements:
+    """This class saves and prints out the information regarding ontology classes and properties
+    that are in the XML but not the ontology."""
+
+    save_path: Path
+    classes: list[list[str]] = field(default_factory=list)
+    properties: list[list[str]] = field(default_factory=list)
+
+    def execute_problem_protocol(self) -> None:
+        """
+        This function is executed if there are any elements in properties or classes.
```

Comment:
That's not true. You can execute it also when there are no elements :-) To be more precise: 

```suggestion
        If there are any elements in properties or classes, this method composes an error message.
```

I agree with Balduin that this method should only compose a string, but not raise an error.